verification/verifier_helpers.go

```diff
@@ -214,10 +214,13 @@ func verifyX509TrustedIdentities(certs []*x509.Certificate, trustPolicy *TrustPo
 
 	var trustedX509Identities []map[string]string
 	for _, identity := range trustPolicy.TrustedIdentities {
-		i := strings.Index(identity, ":")
+		parts := strings.Split(identity, ":")
+		if len(parts) != 2 {
+			return fmt.Errorf("trustpolicy identity `%s` malformed.", identity)
+		}
```

Comment:
what if the identity is '*'?

Reply:
> what if the identity is '*'?

I don't think [spec](https://github.com/notaryproject/notaryproject/blob/main/trust-store-trust-policy-specification.md#trusted-identities-constraints) supports `*` identities
```
The minimal possible value is x509.subject: C=${country}, ST=${state}, O={organization},
```

Reply:
> > what if the identity is '*'?
> 
> I don't think [spec](https://github.com/notaryproject/notaryproject/blob/main/trust-store-trust-policy-specification.md#trusted-identities-constraints) supports `*` identities
> 
> ```
> The minimal possible value is x509.subject: C=${country}, ST=${state}, O={organization},
> ```
According to the spec:
The trustedIdentities list items MUST support a full and partial list of all the attribute types present in [subject DN](https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1.2.6) of x509 certificate. Alternatively, it supports a single item with value *, to indicate that any certificate that chains to the associated trust store (trustStore) is allowed.

Reply:
`if isPresent(wildcard, trustPolicy.TrustedIdentities) {
		return nil
}`
does the '*' check at the top of this func, it's not in the change of this PR so I missed it. LGTM now.